src/test/ui/macros/pub-macro-rules-fail.rs

```diff
@@ -0,0 +1,23 @@
+#![feature(pub_macro_rules)]
+
+#[macro_use]
+mod m {
```

Comment:
I think it would be a good idea to add one of the weird cases from https://github.com/rust-lang/rust/pull/77984 (e.g. `const _: () = { pub macro_rules! mac { () => {} } }`), and verify that the normal scoping rules are applied.